model/search.js

```diff
@@ -0,0 +1,214 @@
+import { getDatetime, getUrl } from '@inrupt/solid-client/thing/get';
+import { DCTERMS } from '@inrupt/vocab-common-rdf';
+import Fuse from 'fuse.js';
+import { itemPath } from '../utils/uris';
+
+export function defaultOptions(keys) {
+  return {
+    includeScore: true,
+    ignoreLocation: true,
+    threshold: 0.3,
+    keys: keys,
+  };
+}
+
+export const FullTextFuseKeys = ['title', 'description', 'text'];
+export const FuseKeys = ['title', 'description'];
+
+function fuseEntryFromGardenItem(item, gardenUrl) {
+  return {
+    title: getTitle(item),
+    description: getDescription(item),
+    uuid: getUUID(item),
+    href:
+      gardenUrl &&
+      itemPath(getCreator(item), HomeSpaceSlug, gardenUrl, getTitle(item)),
+  };
+}
+
+export function fuseEntriesFromGardenItems(items, gardenUrl) {
+  return items.map((item) => fuseEntryFromGardenItem(item, gardenUrl));
+}
+
+async function fullTextFuseEntryFromGardenItem(item, gardenUrl, options) {
+  const noteBodyResourceUrl = item && getNote(item);
+  const noteResource = await getSolidDataset(noteBodyResourceUrl, options);
+  const noteBody = getThing(noteResource, noteBodyResourceUrl);
+  const valueThing = getThing(noteResource, noteBody && getNoteValue(noteBody));
+
+  const value =
+    valueThing &&
+    noteResource &&
+    thingsToArray(valueThing, noteResource, noteThingToSlateObject);
+
+  const valueStr = JSON.stringify(value);
+  const itemEntry = fuseEntryFromGardenItem(item, gardenUrl);
+  return {
+    ...itemEntry,
+    text: valueStr,
+  };
+}
+
+export async function fullTextFuseEntriesFromGardenItems(
+  items,
+  gardenUrl,
+  options
+) {
+  return await Promise.all(
+    items.map((item) =>
+      fullTextFuseEntryFromGardenItem(item, gardenUrl, options)
+    )
+  );
+}
+
+async function setupOrUpdateGardenSearchIndex(garden, fuseIndexUrl, { fetch }) {
+  const gardenUrl = getSourceUrl(garden);
+  const fuseIndexInfo = getThing(garden, fuseIndexUrl);
+  const fuseLastModified =
+    fuseIndexInfo && getDatetime(fuseIndexInfo, DCTERMS.modified);
+  const fuseResp =
+    fuseLastModified &&
+    (await fetch(fuseIndexUrl, {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+      },
+    }));
+  const fuseIndex = fuseResp && Fuse.parseIndex(await fuseResp.json());
+
+  const allItems = getItemAll(garden);
+  if (allItems.length > 0) {
+    let itemsToUpdate = [];
+    if (fuseLastModified) {
+      for (const item of allItems) {
+        if (getDatetime(item, DCTERMS.modified) > fuseLastModified) {
+          itemsToUpdate = [...itemsToUpdate, item];
+        }
+      }
+    } else {
+      itemsToUpdate = allItems;
+    }
+
+    if (itemsToUpdate.length > 0) {
+      const entriesToUpdate = await fullTextFuseEntriesFromGardenItems(
+        itemsToUpdate,
+        gardenUrl,
+        {
+          fetch,
+        }
+      );
+      console.log(`Entries to update: ${entriesToUpdate.length}`);
```

Comment:
this one too?

Reply:
These happen server-side, so I'd prefer to keep them. They are the only way we get any info back from the vercel function logs about what is happening, and folks only see them when they are actively logged into the function logs page (since we don't have a log drain set up)